packages/cli/src/parallel-tests/parallel-tests.handler.ts

```diff
@@ -245,6 +152,18 @@ export const runAllTestsInParallel: (
           })}'`,
           err
         );
+      })
+      .finally(() => {
+        // We only decrement inProgress at the very end,
+        // otherwise another promise may call allTasksDone.resolve() while
+        // we're still saving results
+        --inProgress;
+
+        if (queue.length === 0 && inProgress === 0) {
+          allTasksDone.resolve();
+        }
+
+        process.nextTick(checkNextTask);
```

Comment:
Bug #2: If there was a failure inside onTestFinished then we would fail without continuing to check for the next task